video_creation/final_video.py

```diff
@@ -81,103 +121,136 @@ def make_final_video(
+    # Create a screenshot_width variable to scale the screenshots to the correct size, the calculation is int((W * 90) // 100)
+    # Convert it to a ffmpeg one with iw-
+    screenshot_width = int((W * 45) // 100)
+    audio = ffmpeg.input(f"assets/temp/{reddit_id}/audio.mp3")
+
+    image_clips = list()
+
     image_clips.insert(
         0,
-        ImageClip(f"assets/temp/{reddit_id}/png/title.png")
-        .set_duration(audio_clips[0].duration)
-        .resize(width=screenshot_width)
-        .set_opacity(new_opacity)
-        .crossfadein(new_transition)
-        .crossfadeout(new_transition),
+        ffmpeg.input(f"assets/temp/{reddit_id}/png/title.png")["v"].filter(
+            "scale", screenshot_width, -1
+        ),
     )
+
+    current_time = 0
     if settings.config["settings"]["storymode"]:
+        audio_clips_durations = [
+            float(
+                ffmpeg.probe(f"assets/temp/{reddit_id}/mp3/postaudio-{i}.mp3")[
+                    "format"
+                ]["duration"]
+            )
+            for i in range(number_of_clips)
+        ]
+        audio_clips_durations.insert(
+            0,
+            float(
+                ffmpeg.probe(f"assets/temp/{reddit_id}/mp3/title.mp3")["format"][
+                    "duration"
+                ]
+            ),
+        )
         if settings.config["settings"]["storymodemethod"] == 0:
             image_clips.insert(
                 1,
-                ImageClip(f"assets/temp/{reddit_id}/png/story_content.png")
-                .set_duration(audio_clips[1].duration)
-                .set_position("center")
-                .resize(width=screenshot_width)
-                .set_opacity(float(opacity)),
+                ffmpeg.input(f"assets/temp/{reddit_id}/png/story_content.png").filter(
+                    "scale", screenshot_width, -1
+                ),
             )
+            background_clip = background_clip.overlay(
+                image_clips[1],
+                enable=f"between(t,{current_time},{current_time + audio_clips_durations[1]})",
+                x="(main_w-overlay_w)/2",
+                y="(main_h-overlay_h)/2",
+            )
+            current_time += audio_clips_durations[1]
         elif settings.config["settings"]["storymodemethod"] == 1:
             for i in track(
                 range(0, number_of_clips + 1), "Collecting the image files..."
             ):
                 image_clips.append(
-                    ImageClip(f"assets/temp/{reddit_id}/png/img{i}.png")
-                    .set_duration(audio_clips[i + 1].duration)
-                    .resize(width=screenshot_width)
-                    .set_opacity(new_opacity)
-                    # .crossfadein(new_transition)
-                    # .crossfadeout(new_transition)
+                    ffmpeg.input(f"assets/temp/{reddit_id}/png/img{i}.png")["v"].filter(
+                        "scale", 1080, -1
+                    )
```

Comment:
dont 1080 use ``` (W * 0.8 )```

Reply:
Wouldn't it just be `W` since that's 1080? But thanks for the review!

Reply:
No it will cause trouble in vertical videos because the video after cropping is only 600px